src/mcp_server_uyuni/auth.py

```diff
@@ -0,0 +1,19 @@
+from fastmcp.server.auth import RemoteAuthProvider
+from fastmcp.server.auth.providers.jwt import JWTVerifier
+from pydantic import AnyHttpUrl
+
+
+class AuthProvider(RemoteAuthProvider):
+    def __init__(self, auth_server):
+        verifier = JWTVerifier(
+            jwks_uri=auth_server + "/protocol/openid-connect/certs",
+            issuer=auth_server,
+            audience="mcp-server-uyuni",
+            required_scopes=[],
+        )
+
+        super().__init__(
+            token_verifier=verifier,
+            authorization_servers=[AnyHttpUrl(auth_server)],
+            resource_server_url="http://localhost:8000/mcp", #TODO: Get URL dynamically?
```

Comment:
you can use host and port properties of the FastMCP Server implementation

https://github.com/jlowin/fastmcp/blob/main/src/fastmcp/server/server.py#L162
https://github.com/jlowin/fastmcp/blob/main/src/fastmcp/server/server.py#L163

you will have to pass them to the class constructor, or pass the FastMCP object itself.